@maxc01 a few style and type checks failing:

```
src/gluonts/torch/__init__.py:31:25: F401 [*] `.model.tide.TiDEEstimator` imported but unused
```

```
src/gluonts/torch/model/tide/module.py:292: error: Argument 1 to "sum" has incompatible type "Optional[List[int]]"; expected "Iterable[bool]"  [arg-type]
src/gluonts/torch/model/tide/module.py:301: error: Argument "embedding_dims" to "FeatureEmbedder" has incompatible type "Optional[List[int]]"; expected "List[int]"  [arg-type]
```

--- a/src/gluonts/torch/__init__.py
+++ b/src/gluonts/torch/__init__.py
@@ -16,6 +16,7 @@ __all__ = [
     "PyTorchPredictor",
     "DeepNPTSEstimator",
     "DeepAREstimator",
+    "TiDEEstimator",
     "SimpleFeedForwardEstimator",
     "TemporalFusionTransformerEstimator",
     "WaveNetEstimator",

--- a/src/gluonts/torch/model/tide/estimator.py
+++ b/src/gluonts/torch/model/tide/estimator.py
@@ -121,7 +121,7 @@ class TiDEEstimator(PyTorchLightningEstimator):
         This must be set if ``num_feat_static_cat > 0`` (default: None).
     embedding_dimension
         Dimension of the embeddings for categorical features
-        (default: ``[min(16, (cat+1)//2) for cat in cardinality]``).
+        (default: ``[16 for cat in cardinality]``).
     layer_norm
         Enable layer normalization or not (default: False).
     lr

--- a/src/gluonts/torch/model/tide/module.py
+++ b/src/gluonts/torch/model/tide/module.py
@@ -231,7 +231,7 @@ class TiDEModel(nn.Module):
         num_feat_static_real: int,
         num_feat_static_cat: int,
         cardinality: List[int],
-        embedding_dimension: Optional[List[int]],
+        embedding_dimension: List[int],
         feat_proj_hidden_dim: int,
         encoder_hidden_dim: int,
         decoder_hidden_dim: int,
@@ -253,10 +253,7 @@ class TiDEModel(nn.Module):
         assert num_feat_static_real > 0
         assert num_feat_static_cat > 0
         assert len(cardinality) == num_feat_static_cat
-        assert (
-            embedding_dimension is None
-            or len(embedding_dimension) == num_feat_static_cat
-        )
+        assert len(embedding_dimension) == num_feat_static_cat
 
         self.context_length = context_length
         self.prediction_length = prediction_length
@@ -275,17 +272,9 @@ class TiDEModel(nn.Module):
         self.decoder_output_dim = decoder_output_dim
         self.dropout_rate = dropout_rate
 
-        # TODO: add decomposition like dlinear
-        # self.decomposition = SeriesDecomp(kernel_size)
-
-        # TODO: the bottleneck is here, the flattened feature is too long,
-        # e.g. H=720,L=720,d=4, then the feature is 5760 dim
         self.proj_flatten_dim = (
             context_length + prediction_length
         ) * num_feat_dynamic_proj
-        # TODO: the above issue makes the encoder input dim too high, which is
-        # 720 + 1 (embedded cat dimention) + 5760=6481, if we add TS
-        # embedding, this is even longer
         encoder_input_dim = (
             context_length
             + num_feat_static_real

--- a/src/gluonts/torch/model/tide/module.py
+++ b/src/gluonts/torch/model/tide/module.py
@@ -11,7 +11,7 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-from typing import List, Optional, Tuple
+from typing import List, Tuple
 
 import torch
 from torch import nn

